spec/javascript/packages/document-capture/components/document-side-acuant-capture-spec.jsx

```diff
@@ -0,0 +1,40 @@
+import { DeviceContext, FeatureFlagContext } from '@18f/identity-document-capture';
+import DocumentsStep from '@18f/identity-document-capture/components/documents-step';
+import { render } from '../../../support/document-capture';
+
+describe('DocumentSideAcuantCapture', () => {
+  context('when selfie is _not_ enabled', () => {
+    it('_does_ display a photo upload button', () => {
+      const { queryAllByText } = render(
+        <FeatureFlagContext.Provider value={{ selfieCaptureEnabled: false }}>
+          <DeviceContext.Provider value={{ isMobile: true }}>
+            <DocumentsStep />
+          </DeviceContext.Provider>
+        </FeatureFlagContext.Provider>,
+      );
+
+      const takeOrUploadPictureText = queryAllByText(
+        'doc_auth.buttons.take_or_upload_picture_html',
+      );
+      expect(takeOrUploadPictureText).to.have.lengthOf(2);
+    });
+  });
+
+  context('when selfie _is_ enabled', () => {
+    it('does _not_ display a photo upload button', () => {
+      const { queryAllByText } = render(
+        <FeatureFlagContext.Provider value={{ selfieCaptureEnabled: true }}>
+          <DeviceContext.Provider value={{ isMobile: true }}>
+            <DocumentsStep />
+          </DeviceContext.Provider>
+        </FeatureFlagContext.Provider>,
+      );
+
+      const takePictureText = queryAllByText('doc_auth.buttons.take_picture');
+      expect(takePictureText).to.have.lengthOf(3);
+
+      const notExpectedText = 'doc_auth.buttons.take_or_upload_picture_html';
+      expect(queryAllByText(notExpectedText)).to.be.an('array').that.is.empty;
+    });
+  });
+});
```

Comment:
I felt like more tests might be needed here. And it feels strange that this test is in the `document-side-acuant-capture-spec.jsx` perhaps when that component isn't even directly referred to. But I hit the following issues:

1. `document-side-acuant-capture-spec.jsx` is what determines the logic of whether or not the upload is allowed, so from that standpoint it felt like the best place
2. the other files changed in this PR are mostly passing around the selfieEnabled value and don't change the display at all, and React doesn't promote testing at that level.
3. I tried to write feature tests in `spec/features/idv/doc_auth/document_capture_spec.rb`, but found I couldn't. I could because `perform_in_browser(:mobile)` doesn't actually make the browser mobile, and this doesn't change anything on the desktop flow.

Reply:
@night-jellyfish , the front check whether it's mobile and set a flag. With following mock, looks like we can trick the backend to think it's `mobile`.

While `perform_in_browser(:mobile)` is more like do it in another browser session with nickname `mobile`.

https://github.com/18F/identity-idp/blob/367ee3c7c5dec7942b2aaafad537232901193f49/spec/features/idv/doc_auth/document_capture_spec.rb#L253-L259